src/registry/licensePlate/validationSchema.ts

```diff
@@ -0,0 +1,27 @@
+import type {LicensePlateComponentSchema} from '@open-formulieren/types';
+import {defineMessage} from 'react-intl';
+import {z} from 'zod';
+
+import type {GetValidationSchema} from '@/registry/types';
+
+const LICENSE_PLATE_INVALID_MESSAGE = defineMessage({
+  description: 'Validation error for license plate.',
+  defaultMessage: 'Invalid Dutch license plate',
+});
+
+const getValidationSchema: GetValidationSchema<LicensePlateComponentSchema> = (
+  componentDefinition,
+  intl
+) => {
+  const {key, validate} = componentDefinition;
+  const {required, pattern} = validate;
+
+  let schema: z.ZodFirstPartySchemaTypes = z
+    .string()
+    .regex(new RegExp(pattern), {message: intl.formatMessage(LICENSE_PLATE_INVALID_MESSAGE)});
```

Comment:
I *think* we want to hardcode the pattern here instead of taking it from the component definition, which allows us to retro-actively validate license plates if the regex every changes in the future, without needing to update all existing form definitions in the backend. For type-safety and sanity checks, I'd probably do something like:

```ts
const LICENSE_PLATE_PATTERN: LicensePlateComponentSchema['validate']['pattern'] = '<regex here>';
const LICENSE_PLATE_REGEX = new RegExp(LICENSE_PLATE_PATTERN);
```
so that we have a module constant that only needs to initialize once.